pkg/util/proc/procmanager.go

```diff
@@ -114,7 +116,24 @@ func (p *ProcManager) Shutdown() {
 	p.RUnlock()
 }
 
-func (p *ProcManager) checkProcessExists(binary, procSearchPattern string, policy ProcStartPolicy) (bool, error) {
+// Checks if a process exists. If the restart policy indicates the process should be restarted,
+// the process will be stopped here.
+func (p *ProcManager) checkProcessExists(binary, procSearchPattern string, policy ProcStartPolicy) (shouldStart bool, err error) {
+	// check if this process is already being monitored
+	p.RLock()
+	defer p.RUnlock()
+	if index, proc := p.findMonitoredProc(binary, procSearchPattern); proc != nil {
+		if policy == ReuseExisting {
+			logger.Infof("Process not running, but will be started by the process monitor. %s", procSearchPattern)
```

Comment:
how do we know that the process isn't running at this point?  don't we only know that it's under our management?

also, we say in the log statement that it will be started, then we return "false" for the "shouldStart" return param.  I'm confused about the semantics there...